ui/analyse/src/util.ts

```diff
@@ -336,7 +341,8 @@ export const parseLexicalUci = (uci: string): LexicalUci | undefined => {
 };
 
 export const isOnlyDropsPly = (node: Tree.Node, variantKey: VariantKey, defaultValue: boolean) => {
-  if (variantKey === 'amazons') return node.drops !== undefined && node.drops !== null && node.drops.length > 0;
+  if (variantKey === 'amazons')
+    return node.dropsByRole !== undefined && node.dropsByRole !== null && node.dropsByRole.length > 0;
```

Comment:
```suggestion
    return Array.isArray(node.dropsByRole) && node.dropsByRole.length;
```

I think it should work:
https://www.typescriptlang.org/pt/play/?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBCcQQ0RxkJj3HiQZIZSCuAC8oC43EZzO4AG5FBSqWIyLS8AA5KLQUDsziS-mC6kipgAeWSxAoAC9IAAtb4YdkVJV4AAUAEp5Y0hTSVWrNZBlZwUAajcaACwAJnN5Mt8HcAWg8GYxsNooCQhDTGN1rw9K5TMd3FN-AjeDDghTUaNEug0CTlSNaYz0dV6q1urYeZThZdxdtWodkFNpoFPr9AaDtdLOr1-pRzFiFyNJbtDd7nH7mC9xRwzQ8ZC8iHgfQYwPYNC+bjMzHYDGYkH4sGg6kSK-gehEXgARABRVYMRALfSXpLYy-gByoADkwjFkAkl6CFRlS+Mh3kEfctEgI5UEcBRXGkPBQCcRcdGGLB9BQBBEGhYRLg3CQl2iEQDA4bEFiYVgOB4IISiAA